frontend/src/app/hierarchy-viewer/hierarchy-viewer.component.ts

```diff
@@ -19,32 +22,88 @@ export class HierarchyViewerComponent implements OnDestroy {
     subscriptions: Subscription[] = [];
     hierarchies$?: Observable<HierarchyListItem[]>;
 
-    constructor(private store: Store<HierarchyState>) {
-        const sub = this.store.select(getSelectedHierarchy)
+    constructor(public dialog: MatDialog, private store: Store<HierarchyState>, private renderer: Renderer2) {
+        const selectedHierarchySub = this.store.select(getSelectedHierarchy)
             .subscribe(hierarchy => {
                 if (hierarchy) {
-                    this.selectedHierarchy = hierarchy;              
+                    this.selectedHierarchy = hierarchy;
+                    return;
                 }
             });
-        this.subscriptions.push(sub);
-    
+        this.subscriptions.push(selectedHierarchySub);
+         
+
         this.hierarchies$ = this.store.select(getHierarchies);
     }
 
     ngOnDestroy(): void {
         this.subscriptions.forEach(sub => sub.unsubscribe());
     }
 
-    createHierarchy() {
-        this.selectedHierarchy = null;
-        //This will go away eventually. We will replace it with creating hierarchies
-        //from scratch
-        this.store.dispatch(createHierarchy({hierarchy: RRRHierarchy}));
-        //this.store.dispatch(createHierarchy({hierarchy: SimpleHierarchy}))
+    deleteHierarchy() {
+        if (this.selectedHierarchy) {
+            this.store.dispatch(deleteHierarchy({ hierarchyId: this.selectedHierarchy.id }));
+        }
     }
 
     onSelect(event: MatSelectChange) {
         this.selectedHierarchy = null;
-        this.store.dispatch(setSelectedHierarchy({selectedHierarchyId: event.value}));
+        this.store.dispatch(setSelectedHierarchy({ selectedHierarchyId: event.value }));
+    }
+
+    createHierarchyDialog() {
+        const dialogRef = this.dialog.open(CreateHierarchyDialogComponent, {
+            data: { name: '', description: '' }
+        });
+
+        dialogRef.afterClosed().subscribe(result => {
```

Comment:
I don't think we should be waiting for the dialog to close before dispatching the create/delete actions. The dialog should handle that.

Reply:
I'll look into this. It''ll probably require some logic change for creating the tree.